diff --git a/node_modules/react-native/Libraries/Components/View/View.js b/node_modules/react-native/Libraries/Components/View/View.js
index a7177fc..edcce3b 100644
--- a/node_modules/react-native/Libraries/Components/View/View.js
+++ b/node_modules/react-native/Libraries/Components/View/View.js
@@ -23,10 +23,16 @@ import {use} from 'react';
  *
  * @see https://reactnative.dev/docs/view
  */
-export default component View(
+const View: component(
   ref?: React.RefSetter<React.ElementRef<typeof ViewNativeComponent>>,
   ...props: ViewProps
-) {
+) = ({
+  ref,
+  ...props
+}: {
+  ref?: React.RefSetter<React.ElementRef<typeof ViewNativeComponent>>,
+  ...ViewProps,
+}) => {
   const hasTextAncestor = use(TextAncestorContext);
 
   let actualView;
@@ -212,4 +218,7 @@ export default component View(
     );
   }
   return actualView;
-}
+};
+
+View.displayName = 'View';
+export default View;
\ No newline at end of file
